### DISCUSSION TITLE
Feature request: Implement "First start" check

### DISCUSSION THREAD

**Discussion by WhiteChairFromIkea** [doublecmd/doublecmd#1155] [Suggestions]
I am a windows user, TC is unknown to me. It was very hard to start using DC at first, because keyboard shortcuts were totally different than windows user is expecting (F5 means refresh, not file operation for windows user!). I believe many potential users will run away from DC with such a hard/different usage start, the need to go through preferences to make GUI more simple, not TC-like.

I propose implementing startup check upon DC launch:
1) Check if this is a first time launch in both portable and installed mode (check if some configuration file is null);
2) In case this is a first time launch, throw a graphical dialog asking:

```
Double commander is launched for the first time. Which GUI and keyboard shortcuts to use?
"Total Commander-like" or "Classic explorer-like"
```
 .. offer two options, showing screenshots and hotkey descriptions:
a) a screenshot of DC GUI as is shipped right now: lower action toolbar (or how is it called in TC) on, F5 is assigned to some file operation...;    
b) Windows explorer screenshot (on Windows, other default explorer from ubuntu, if run in Linux), Copy file = CTRL + C shortcut and so on.

According to user choice, configure GUI properly (TC like, or windows explorer-like), also use corresponding `hotkey.scf`.

This issue is tightly related to https://github.com/doublecmd/doublecmd/issues/166

```
Double Commander
Version: 1.0.0 alpha
Revision: 9890
Commit: 5b000ea17
Build date: 2021/08/22
Lazarus: 2.2.0.2
FPC: 3.0.4
Platform: x86_64-Win64-qt5
OS version: Windows 10 Home 20H2 x86_64

Widgetset library: Qt 5.13.0, libQt5Pas 5.6.1
```

**Comment by Skif-off** [comment]
"Total Commander-like" shortcuts? Some of them are older than Total Commander :) "Norton Commander-like"?
And this is really "classic".
https://en.wikipedia.org/wiki/File_manager#Orthodox_file_managers

P.S. Hot keys can be configured once and use long, very long. Why do it every day and suffer from it?

**Comment by WhiteChairFromIkea** [comment]
> P.S. Hot keys can be configured once and use long, very long. Why do it every day and suffer from it?

What do you mean? There are some applications which runs configuration wizard if detected first time run after "fresh installation". As an examples: MS Visual studio does this (asks which shortcuts and GUI configuration to use: C#-related, or C++-related), also kicad.exe (offer to migrate libraries from previous version if detected, asks for which video engine to use).

**Comment by alexx2000** [comment]
It is not correct to compare DC with Explorer. It is two different file managers concepts. And each concept has its own default hotkeys. So current default hotkeys are "classic" (commander style).

**Comment by WhiteChairFromIkea** [comment]
After turning many TC-like features OFF DC becomes convenient file manager, like FreeCommander or nemo, you can see the screenshots of my explorers, there is no difference in look with Freecommander:    
![5529R](https://user-images.githubusercontent.com/46539653/131093296-a5746861-b0bc-4fb1-bdec-0fa603dd5583.png)
![5408V](https://user-images.githubusercontent.com/46539653/131093247-77c97d46-0f72-4b82-8a5a-e31f826f5da0.png)

I stopped using FC because it is damn slow, and not opensource.

DC looks the same as other managers, but feels completely different at the beginning. I believe many potential users does not start using it after trial period. It is hard to make even simple actions (refresh fiew with F5) with stock configuration.

Compared to Free commander, DC does not support good looking file tree, favorite tree (but it is superseded by directory hotlist), and less search capability (duplicate search by MD5). But DC has many superior features which can be found here: https://sourceforge.net/projects/doublecmd/reviews#

Shipping two presets (regular-user-like and tc-dos-user-like) would attract many new users. Only initial GUI and Hotkey settings makes DC different than others.

**Comment by alexx2000** [comment]
> favorite tree (but it is superseded by directory hotlist), and less search capability (duplicate search by MD5)

DC has powerful "Tree View Menu" which can be used with directory hotlist, favorite tabs and navigation history, see [documentation](https://skif-off.github.io/doublecmd.github.io/doc/en/configuration.html#ConfigTreeMenu).
DC has duplicate search by content or hash (not MD5 but much faster BLAKE2), see [documenation](https://skif-off.github.io/doublecmd.github.io/doc/en/findfiles.html#advanced).

**Comment by WhiteChairFromIkea** [comment]
Thank you for pointing me to duplicate search, I missed that item, , this is a great news for me :)!

I will give a 2nd try for "Tree view menu". I tried it once at the beginning, and it seems like a duplicate of graphical "Directory hotlist" which is more convenient for me. "Tree view" shows too much information, lists all special dirs (environment variables) for some reason...
